packages/babel-helpers/src/helpers.ts

```diff
@@ -812,8 +818,13 @@ helpers.iterableToArrayLimitLoose = helper("7.0.0-beta.0")`
     if (_i == null) return;
 
     var _arr = [];
-    for (_i = _i.call(arr), _step; !(_step = _i.next()).done;) {
-      _arr.push(_step.value);
+    _i = _i.call(arr);
+    if (i === 0) {
+      if (Object(_i) !== _i) return;
+      return _arr;
+    }
+    for (var _s; !(_s = _i.next()).done;) {
+      _arr.push(_s.value);
       if (i && _arr.length === i) break;
     }
```

Comment:
Since the loose helper does not need to be 100% spec compliant, is this enough to avoid infinite loops?
```js
for (_i = _i.call(arr), _step; arr.length < i && !(_step = _i.next()).done;) {
  _arr.push(_step.value);
}
```

Reply:
The variable declaration of `_step` seems to be missing in the original implementation (put it in the first expression of `for` seems to be useless?), so I think the correct code should be this:

```js
var _s;
for (_i = _i.call(arr); arr.length < i && !(_s = _i.next()).done; ) {
  _arr.push(_s.value);
}
```

Committed in 4f951f3.